src/corehost/common/pal.unix.cpp

```diff
@@ -301,12 +301,73 @@ bool pal::get_global_dotnet_dirs(std::vector<pal::string_t>* recv)
 
 bool pal::get_dotnet_self_registered_dir(pal::string_t* recv)
 {
-    // No support for global directories in Unix.
-    return false;
+    recv->clear();
+
+    //  ***Used only for testing***
+    pal::string_t environment_override;
+    if (pal::getenv(_X("_DOTNET_TEST_GLOBALLY_REGISTERED_PATH"), &environment_override))
```

Comment:
I thought you were worried about a secutiry hole with this.

Reply:
Yes - but I didn't want to have one PR solving two separate issues... and I need a test-only thing for this feature. So I kept doing it the "old" way... will fix this in a separate PR for all cases where we do this.